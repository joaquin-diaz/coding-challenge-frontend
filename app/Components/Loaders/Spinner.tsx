import * as React from 'react';
import styled from '@emotion/styled';

interface SpinnerProps {
  color: string;
}

const SpinnerStyles = styled.div<SpinnerProps>`
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
  transform: scale(0.4);

  div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 51px;
    height: 51px;
    margin: 6px;
    border: 6px solid ${props => props.color};
    border-radius: 50%;
    animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: ${props => props.color} transparent transparent transparent;
  }

  div:nth-child(1) {
    animation-delay: -0.45s;
  }

  div:nth-child(2) {
    animation-delay: -0.3s;
  }

  div:nth-child(3) {
    animation-delay: -0.15s;
  }

  @keyframes lds-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const Spinner: React.FC<SpinnerProps> = props => {
  return (
    <SpinnerStyles {...props}>
      <div />
      <div />
      <div />
      <div />
    </SpinnerStyles>
  );
};

export default Spinner;
