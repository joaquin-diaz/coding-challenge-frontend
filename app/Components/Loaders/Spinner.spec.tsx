import * as React from 'react';

import Spinner from './Spinner';
import { mount } from 'enzyme';

describe('<Spinner />', () => {
  it('should render', () => {
    const wrapper = mount(<Spinner color="black" />);
    expect(wrapper).toMatchSnapshot();
  });
});
