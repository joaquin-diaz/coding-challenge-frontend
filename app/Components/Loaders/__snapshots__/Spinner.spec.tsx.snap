// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render 1`] = `
.emotion-0 {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
  -webkit-transform: scale(0.4);
  -ms-transform: scale(0.4);
  transform: scale(0.4);
}

.emotion-0 div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 51px;
  height: 51px;
  margin: 6px;
  border: 6px solid black;
  border-radius: 50%;
  -webkit-animation: lds-ring 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: lds-ring 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  border-color: black transparent transparent transparent;
}

.emotion-0 div:nth-of-type(1) {
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.emotion-0 div:nth-of-type(2) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.emotion-0 div:nth-of-type(3) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

<Spinner
  color="black"
>
  <Styled(div)
    color="black"
  >
    <div
      className="emotion-0"
      color="black"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </Styled(div)>
</Spinner>
`;
